src/server/pkg/storage/chunk/limit_objclient.go

```diff
@@ -0,0 +1,74 @@
+package chunk
+
+import (
+	"context"
+	io "io"
+	"math"
+
+	"github.com/pachyderm/pachyderm/src/server/pkg/obj"
+	"golang.org/x/sync/semaphore"
+)
+
+var _ obj.Client = limitedObjClient{}
+
+type limitedObjClient struct {
+	obj.Client
+	writersSem *semaphore.Weighted
+	readersSem *semaphore.Weighted
+}
+
+func newLimitedObjClient(client obj.Client, maxReaders, maxWriters int) limitedObjClient {
```

Comment:
I think it would be better to operate on a pointer to a `limitedObjClient`.

Reply:
okay can change it. I don't really have any strong intuitions about which is more performant.  Essentially following one pointer to get to 3 rather than copying 3 around.

Reply:
I think the performance difference doesn't matter in this case, I just prefer it for stylistic consistency.